model_interaction.py

```diff
@@ -0,0 +1,88 @@
+from langchain_community.document_loaders import PDFPlumberLoader
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from langchain_core.vectorstores import InMemoryVectorStore
+from langchain_ollama import OllamaEmbeddings
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_ollama.llms import OllamaLLM
+
+import re
+
+# Initialize embeddings and model
+embeddings = OllamaEmbeddings(model='nomic-embed-text')
+model = OllamaLLM(model="deepseek-r1")
+
+# Initialize vector store
+vector_store = None
+
+def load_pdf(file_path):
+    """Load the content of the PDF using PDFPlumberLoader."""
+    try:
+        loader = PDFPlumberLoader(file_path)
+        return loader.load()
+    except Exception as e:
+        print(f"Error loading PDF: {e}")
+        return None
+
+
+def split_text(documents):
+    """Split the documents into smaller chunks for indexing."""
+    text_splitter = RecursiveCharacterTextSplitter(
+        chunk_size=1000, chunk_overlap=200, add_start_index=True
+    )
+    return text_splitter.split_documents(documents)
+
+
+def index_docs(documents):
+    """Index the documents in the vector store."""
+    global vector_store
+    vector_store = InMemoryVectorStore(embeddings)
+    vector_store.add_documents(documents)
+
+
+# Retrieve simlarity search results
+def retrieve_docs(query):
+    """Retrieve relevant documents based on the query."""
+    return vector_store.similarity_search(query)
+
+# Returns LLMs answer based on question and vector search query return 
+def answer_question(question, related_documents):
+    """Generate an answer to the question using the retrieved documents."""
```

Comment:
```suggestion
    """Generate an answer to the question using the retrieved documents."""
      if not related_documents:
        return "No relevant information found."  
```
Need to handle empty string.